File: app/src/main/java/com/example/solar_alarm/Data/Daos/SolarAlarmDao.java
package com.example.solar_alarm.Data.Daos;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.solar_alarm.Data.Tables.SolarAlarm;

import java.util.List;

@Dao
public interface SolarAlarmDao
{
    @Insert
    void Insert(SolarAlarm solarAlarm);

    @Query("SELECT * FROM SolarAlarm ORDER BY Name")
    LiveData<List<SolarAlarm>> getAll();

    @Query("SELECT EXISTS(SELECT * FROM SolarAlarm WHERE Name = :name AND LocationId = :locationId)")
    boolean isSolarAlarmNameLocationIDPairExists(String name, int locationId);

    @Update
    void Update(SolarAlarm solarAlarm);

    @Delete
    void delete(SolarAlarm solarAlarm);
}

File: app/src/main/java/com/example/solar_alarm/Data/Daos/SolarAlarmDao.kt
package com.example.solar_alarm.data.daos

// Repair: Fixed broken package/import lines
import com.example.solar_alarm.data.tables.SolarAlarm
import androidx.room.*
import com.example.solar_alarm.data.tables.SolarAlarmWithDetails
import com.example.solar_alarm.data.tables.SolarTime
import kotlinx.coroutines.flow.Flow

@Dao
abstract class SolarAlarmDao : BaseDao<SolarAlarm>
{
    @Query("SELECT * FROM SolarAlarm ORDER BY Name")
    abstract fun getAll(): Flow<List<SolarAlarm>>

    @Transaction
    @Query("SELECT * FROM SolarAlarm ORDER BY Name")
    abstract fun getAllWithDetails(): Flow<List<SolarAlarmWithDetails>>

    @Query("SELECT * FROM SolarAlarm WHERE Id = :id")
    abstract suspend fun getById(id: Int): SolarAlarm?

    @Query("SELECT EXISTS(SELECT * FROM SolarAlarm WHERE Name = :name AND LocationId = :locationId)")
    abstract suspend fun isSolarAlarmNameLocationIdPairExists(name: String?, locationId: Int): Boolean
}